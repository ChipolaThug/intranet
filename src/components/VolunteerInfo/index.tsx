import { memo } from "react"

import { Volunteer } from "../../services/volunteers"
import styles from "./styles.module.scss"

interface Props {
    item: Volunteer
}

const VolunteerInfo = ({ item }: Props) => (
    <div className={styles.VolunteerCard}>
        <h4>Volunteer Info</h4>
        <ul>
            <li>Pr√©nom: {item.firstname}</li>
            <li>Nom: {item.lastname}</li>
        </ul>
    </div>
)

export default memo(VolunteerInfo)
