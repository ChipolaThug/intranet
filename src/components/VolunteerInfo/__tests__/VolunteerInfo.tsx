/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react"
import { MemoryRouter } from "react-router-dom"

import VolunteerInfo from "../index"

describe("<VolunteerInfo />", () => {
    it("renders", () => {
        const tree = render(
            <MemoryRouter>
                <VolunteerInfo
                    item={{
                        id: 1,
                        firstname: "Aupeix",
                        lastname: "Amélie",
                        email: "pakouille.lakouille@yahoo.fr",
                        mobile: "0675650392",
                        photo: "images/volunteers/$taille/amélie_aupeix.jpg",
                        food: "Végétarien",
                        adult: 1,
                        privileges: 0,
                        active: 0,
                        created: new Date(0),
                        password1: "$2y$10$fSxY9AIuxSiEjwF.J3eXGubIxUPkdq9d5fqpbl8ASimSjNj4SR.9O",
                        password2: "$2y$10$fSxY9AIuxSiEjwF.J3eXGubIxUPkdq9d5fqpbl8ASimSjNj4SR.9O",
                    }}
                />
            </MemoryRouter>
        ).container.firstChild

        expect(tree).toMatchSnapshot()
    })
})
