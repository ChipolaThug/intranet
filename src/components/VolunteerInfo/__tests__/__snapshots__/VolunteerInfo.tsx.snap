// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VolunteerInfo /> renders 1`] = `
<div
  class="VolunteerCard"
>
  <h4>
    Volunteer Info
  </h4>
  <ul>
    <li>
      Prénom: 
      Aupeix
    </li>
    <li>
      Nom: 
      Amélie
    </li>
  </ul>
</div>
`;
